### PR TITLE
RD2: Telemetry for RD2_Enabled

### PR DESCRIPTION
**Pull request by npsp-reedestockton**
Added telemetry to determine whether Enhanced Recurring Donations is enabled.
# Critical Changes

# Changes

- We added Feature Telemetry for upcoming Recurring Donation enhancements. No data should be collected until this functionality is released at a later date. NPSP Feature Telemetry collects feature and usage metrics when customers use our products. We use this information to understand the level of usage and adoption of various NPSP features. No personally identifiable information is collected through NPSP Feature Telemetry. Read more about NPSP Telemetry here: https://powerofus.force.com/s/article/NPSP-Feature-Telemetry. 

# Issues Closed

# New Metadata
Feature Parameters:  

- IsEnabled_RecurringDonations2.featureParameterBoolean

# Deleted Metadata

### CODE REVIEW

**Comment by npsp-reedestockton** [comment]
@force2b The line below my TODO comment on line 124 of src/classes/UTIL_OrgTelemetry_TEST.cls causes a test failure.  Either I am mishandling something, or this line is no longer a valid test of the method.  Would appreciate your guidance.

File: src/classes/UTIL_OrgTelemetry_SVC.cls [hunk 1]
```diff
@@ -73,6 +73,7 @@ public without sharing class UTIL_OrgTelemetry_SVC {
         IsEnabled_BDICMT,
         IsEnabled_CustomizableRollups,
         IsEnabled_DefaultGAU,
+        RD2_Enabled,
```
[the remark this answers is not in the document]
Comment (npsp-reedestockton) [reply src/classes/UTIL_OrgTelemetry_SVC.cls]: Name changed to IsEnabled_RecurringDonations2, as requested.

File: src/objects/npe03__Recurring_Donations_Settings__c.object [hunk 2]
```diff
@@ -1,5 +1,13 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <CustomObject xmlns="http://soap.sforce.com/2006/04/metadata">
+    <fields>
+        <fullName>IsRecurringDonations2Enabled__c</fullName>
```
[the remark this answers is not in the document]
Comment (npsp-reedestockton) [reply src/objects/npe03__Recurring_Donations_Settings__c.object]: IsEnabled_RecurringDonations2 was our eventual decision.

**Comment by sam-knox** [comment]
**lurch: attach W-035303

**Comment by LurchTheButler** [comment]
Tracking <a href="https://foundation.my.salesforce.com/a2x1E000001C9P9QAK" target="_blank">W-035303</a>

**Comment by npsp-reedestockton** [comment]
@sam-knox There is overlap between this work item and W-035558.  I have tried to reflect changes, but it is worth reviewing labels to make sure latest changes are not overwritten.

File: src/classes/UTIL_OrgTelemetry_SVC.cls [hunk 3]
```diff
@@ -232,6 +233,9 @@ public without sharing class UTIL_OrgTelemetry_SVC {
     private void handleNPSPSettings() {
         STG_SettingsService stgSvc = new STG_SettingsService();
         Addr_Verification_Settings__c addrVerifSettings = UTIL_CustomSettingsFacade.getAddressVerificationSettings();
+        if (Test.isRunningTest()) {
```
Comment (RadSFDeveloper) [inline src/classes/UTIL_OrgTelemetry_SVC.cls]: We can add the configuration in the unit test instead using `Test.isRunningTest()` in the code here.

Comment (npsp-reedestockton) [reply src/classes/UTIL_OrgTelemetry_SVC.cls]: Moved configuration as suggested and eliminated unnecessary Test.isRunningTest()

File: src/objects/npe03__Recurring_Donations_Settings__c.object [hunk 4]
```diff
@@ -1,5 +1,15 @@
...
+        <defaultValue>false</defaultValue>
+        <description>This option enables Enhanced Recurring Donations. WARNING: When you enable this option, NPSP changes your existing Recurring Donations records and the Recurring Donations object to match the data model for Enhanced Recurring Donations.</description>
```
Comment (force2b) [inline src/objects/npe03__Recurring_Donations_Settings__c.object]: FYI - The helptext and descriptions were modified again in Meena's. PR. Should be final now.

Comment (npsp-reedestockton) [reply src/objects/npe03__Recurring_Donations_Settings__c.object]: @force2b Requested changes have been made to alphabetize fields and update description/helptext.

File: src/package.xml [hunk 5]
```diff
@@ -1192,6 +1192,7 @@
         <members>npe03__Recurring_Donation__c.Always_Use_Last_Day_Of_Month__c</members>
         <members>npe03__Recurring_Donation__c.Day_of_Month__c</members>
         <members>npe03__Recurring_Donations_Settings__c.Recurring_Donation_Batch_Size__c</members>
+        <members>npe03__Recurring_Donations_Settings__c.IsRecurringDonations2Enabled__c</members>
```
Comment (force2b) [inline src/package.xml]: Darn, I missed that this one is not in alphabetical order. One last change. Sorry.

**Comment by npsp-reedestockton** [comment]
@force2b package.xml alphabetization change made.

**Comment by Julian88Tex** [comment]
@npsp-reedestockton it appears there are merge conflicts on this branch. Would you mind resolving those first so I know that they don't interfere with accurately testing? Thank you!

**Comment by Julian88Tex** [comment]
Since this was already merged, I will do my checking on the feature/RD2 branch. If that I'll checks out I'll come back here and comment. If now, I'll let the Reed(e)s know: @npsp-reedestockton and @davidmreed .

**Comment by Julian88Tex** [comment]
Confirmed that this code has been successfully added to master and can be pushed to a scratch which is about as far as I can go in testing this ticket until a package is created. I've noted this on our work item so we can consider ways to make sure this is tested once RD2 is merged into master.

**Comment by davidmreed** [comment]
@Julian88Tex I reverted the merge by request.

**Comment by Julian88Tex** [comment]
@davidmreed cool.... new PR created? @npsp-reedestockton ?

**Pull request merged**